@here/harp-mapview/test/DecodedTileHelpersTest.ts

```diff
@@ -187,29 +194,43 @@ describe("DecodedTileHelpers", function () {
             };
 
             const callbackSpy = sandbox.spy();
+            let onImageLoad: ((image: HTMLImageElement) => void) | undefined;
             sandbox
                 .stub(THREE.ImageLoader.prototype, "load")
                 .callsFake((url, onLoad?, _onProgress?, _onError?) => {
                     assert.equal(url, technique.map);
-                    onLoad?.(fakeImageElement);
+                    onImageLoad = onLoad;
                     return fakeImageElement;
                 });
-            let material: MapMeshStandardMaterial | undefined;
-            await new Promise<void>((resolve, _) => {
-                material = createMaterial(
-                    rendererCapabilities,
-                    {
-                        technique,
-                        env
-                    },
-                    callbackSpy
-                ) as MapMeshStandardMaterial;
-                resolve();
+
+            const material = createMaterial(
+                rendererCapabilities,
+                {
+                    technique,
+                    env
+                },
+                callbackSpy
+            ) as MapMeshStandardMaterial;
+            assert.isDefined(onImageLoad);
+            assert.isTrue(callbackSpy.called);
+
+            // Check that texture promised is not yet resolved.
+            let textureLoaded = false;
+            const texturePromise = callbackSpy.firstCall.args[0];
+            texturePromise.then(() => {
```

Comment:
I guess this works because of the `await wait` below, but to be safe, you should `await` here too...? Or put the rest of the test in the `.then()` ?

Reply:
Wait... not here, that would break the test, but before `await wait` ?

Reply:
done, fixed similar cases in other tests as well.